backend/dataall/modules/metadata_forms/services/metadata_form_service.py

```diff
@@ -37,6 +38,42 @@ def validate_create_field_params(data):
             raise exceptions.RequiredParameter('glossaryNodeUri')
 
 
+class MetadataFormAccessService:
+    @staticmethod
+    def is_owner(uri):
+        context = get_context()
+        with context.db_engine.scoped_session() as session:
+            return MetadataFormRepository.get_metadata_form_owner(session, uri) in context.groups
+
+    @staticmethod
+    def can_perform(action: str):
+        def decorator(f):
+            @wraps(f)
+            def check_permission(*args, **kwds):
+                uri = kwds.get('uri')
+                if not uri:
+                    raise KeyError(f"{f.__name__} doesn't have parameter uri.")
+
+                if MetadataFormAccessService.is_owner(uri):
+                    return f(*args, **kwds)
+                else:
+                    raise exceptions.UnauthorizedOperation(
+                        action=action,
+                        message=f'User {get_context().username} is not the owner of the metadata form {uri}',
+                    )
+
+            return check_permission
+
+        return decorator
+
+    @staticmethod
+    def get_user_role(uri):
+        if MetadataFormAccessService.is_owner(uri):
+            return MetadataFormUserRoles.Owner.value
+        else:
+            return MetadataFormUserRoles.User.value
+
+
 class MetadataFormService:
     @staticmethod
     def create_metadata_form(data):
```

Comment:
should creating metadata forms be able to be restricted via a TenantPermission enforced by DA Admins?

Reply:
Oh, I see, what do you mean. I'll save it for another PR, I think.
Right now I found a bug in Admin Setting panel

Reply:
Bug fixed, tennant permissions added